### ISSUE TITLE
CI: Improve documentation around replicating CI locally

### ISSUE THREAD

**Issue by m3dwards** [bitcoin/bitcoin#31199]
I've received some feedback that replicating CI jobs locally isn't straightforward. Perhaps the documentation can be improved in this area?

@maflcko I understand you had some improvements in mind? Happy to also work on this.

**Comment by hodlinator** [comment]
PR is live: #33887

**Comment by maflcko** [comment]
> Another related idea was to move the CC/CXX/SANITZER/... build options into a cmake preset, which could make it easier to reproduce some CI failures. Ref: https://bitcoin-irc.chaincode.com/bitcoin-core-dev/2024-09-05#1050658 and some more discussion in [#30871 (comment)](https://github.com/bitcoin/bitcoin/pull/30871#issuecomment-2344047118)

this was moved to dev-mode in https://github.com/bitcoin/bitcoin/pull/33824, so maybe this is already a step in the right direction?

**Comment by fanquake** [comment]
@m3dwards can you advise if this can be closed after #33887. Or clarify in the description what else should be done.

**Comment by maflcko** [comment]
Yeah, I think this can be closed for now, but I'd be excited to see new issues, if there are still questions or hard CI edges. Also, happy to review more pull requests around CI docs.
